### ISSUE TITLE
YAML parser must not split keys

### ISSUE THREAD

**Issue by mnlipp** [TheElectronWill/night-config#146]
When reading something like 
```yaml
"key.with.dots": 42
```
nested configs are created. There is absolutely no reason to do this as any structure information is explicit when using YAML. (Contrary to TOML where a dot in a key indeed specifies a next level.)

(Similar to https://github.com/TheElectronWill/night-config/issues/121.)

**Comment by TheElectronWill** [comment]
Could I see your code? I cannot reproduce the issue with the following config: `test.yml`
```yml
not.a.subconfig = works
```

```java
FileConfig conf = FileConfig.of("test.yml");
conf.load();
System.out.println(conf); // WriteAsyncFileConfig:SimpleConfig:{not.a.subconfig = works}
```

**Comment by mnlipp** [comment]
> not.a.subconfig = works

Unless I'm very much mistaken, that's not YAML. This is:
```yaml
not.a.subconfig: works
```

Does that also work?

**Comment by TheElectronWill** [comment]
yep sorry, that was a type :smile:
I've added the line to yaml/test.yaml and yes it works.
```yml
sub: {'null': null, nullObject: null}
'null': null
string: this is a string
not.a.subconfig: works
nullObject: null
list: [10, 12]
enum: !!com.electronwill.nightconfig.core.BasicTestEnum 'A'
objectList:
- {foo: bar}
- {baz: true}
```

```java
assertEquals("works", conf.<String>get(Arrays.asList("not.a.subconfig")));
```

**Comment by mnlipp** [comment]
Okay, the better. I had somehow hoped that it works for YAML if it works for JSON. 

I'm using night-config [here](https://github.com/mnlipp/jgrapes/blob/master/org.jgrapes.util/src/org/jgrapes/util/YamlConfigurationStore.java). Reconsidering how I use it, my fault is obvious. I had by now forgotten that due to the lack of OSGi compliance, I had to unpack night-config into my own jar. So I wasn't using the updated version when I tried things today.

Sorry for the noise.
